
version: '2.1'

volumes:
  pg-backup-data-dir:
  pg-data-dir:

services:

  pg_backup:
    image: kartoza/postgis:14-3.1
    restart: 'always'
    volumes:
      - ./setup-db.sql:/docker-entrypoint-initdb.d/setup-db.sql
    environment:
      - POSTGRES_DB=gis,data
      - POSTGRES_USER=docker
      - POSTGRES_PASS=docker
    healthcheck:
      interval: 60s
      timeout: 30s
      retries: 3
      test: "pg_isready"

  pg_restore:
    image: kartoza/pg-backup:manual-build
    restart: 'always'
    volumes:
      - pg-backup-data-dir:/backups
      - ./tests:/tests
      - ../utils:/lib/utils
    environment:
      - DUMPPREFIX=PG_gis
      - POSTGRES_HOST=pg_backup
      - POSTGRES_USER=docker
      - POSTGRES_PASS=docker
      - POSTGRES_PORT=5432
      - TARGET_DB=data
      - TARGET_ARCHIVE=/backups/latest.gis.dmp
      - WITH_POSTGIS=1
      - ARCHIVE_FILENAME=latest
    depends_on:
      pg_backup:
        condition: service_healthy
    healthcheck:
      interval: 60s
      timeout: 30s
      retries: 3
      test: "pg_isready"
